### PR TITLE
Fix #4650, ignore Renovate commits in generate-commit-log script

### LINKED ISSUE #4650: generate-commit-log script should ignore Renovate commits
The Renovate commit messages have consistent patterns, and we should ignore them when generating commit logs.

### PR DESCRIPTION
**Pull request by testeaxeax**
Fix #4650

### CODE REVIEW

File: bin/generate-commit-log [hunk 1]
```diff
@@ -11,6 +11,8 @@ IGNORE_COMMITS = [
     "^chore",
     "^pontoon",
     "UTF.*Pontoon",
+    "^Update .* to .*",
```
Comment (chenba) [inline bin/generate-commit-log]: This regex is too permissive.  But there's a better way.  When the script go through the log messages, it builds a dictionary for each.  So `message.get("from")` will give you the author of the commit, and you could match that against renovate.

**Comment by testeaxeax** [comment]
Thanks for the help

File: bin/generate-commit-log [hunk 2]
```diff
@@ -60,6 +60,8 @@ for message in messages:
         if re.search(pat, sub, re.I):
             should_skip = True
             break
+    if message.get("from") == "Renovate Bot <bot@renovateapp.com>":
+        should_skip = True
```
Comment (chenba) [inline bin/generate-commit-log]: This works great.  But--and it's nit-ish--let's move this check up to be the first thing in the loop, and `continue` when true.  That way we'll just skip subject regex matching.

**Comment by testeaxeax** [comment]
@chenba I updated the commit.

**Comment by chenba** [comment]
@testeaxeax Thanks!

**Pull request merged**